Assets/HoloToolkit/Boundary/Scripts/SceneContentAdjuster.cs

```diff
@@ -7,40 +10,73 @@
 using UnityEngine.VR;
 #endif
 
-public class SceneContentAdjuster : MonoBehaviour
+namespace HoloToolkit.Unity.Boundary
 {
-    private Vector3 lastFloorHeight;
-    private float floorHeightOffset = 1f;
-
-    private void Awake()
+    public class SceneContentAdjuster : MonoBehaviour
     {
-#if UNITY_2017_2_OR_NEWER
-        if (Application.isEditor && XRDevice.isPresent)
+        private int frameWaitHack = 0;
+
+        [SerializeField]
+        [Tooltip("Optional container object reference. If null, this script will move the object it's attached to.")]
+        private Transform containerObject;
+
+        [SerializeField]
+        [Tooltip("Select this if the container should be placed in front of the head on app launch in a room scale app.")]
+        private bool alignWithInitialHeadHeight = true;
+
+        [SerializeField]
+        [Tooltip("Use this to set the desired position of the container in a stationary app.")]
+        private Vector3 stationarySpaceTypePosition = Vector3.zero;
+
+        [SerializeField]
+        [Tooltip("Use this to set the desired position of the container in a room scale app, if alignWithInitialHeadHeight has not been set.")]
+        private Vector3 roomScaleSpaceTypePosition = Vector3.zero;
+
+        private void Awake()
         {
-            lastFloorHeight.y = floorHeightOffset;
-            transform.position = lastFloorHeight;
-        }
+            if (containerObject == null)
+            {
+                containerObject = transform;
+            }
+
+#if UNITY_2017_2_OR_NEWER
+            if (!XRDevice.isPresent)
 #else
-        if (VRDevice.isPresent)
-        {
-            Destroy(this);
-        }
+            if (true)
 #endif
-    }
+            {
+                Destroy(this);
+            }
+            else
+            {
+                StartCoroutine(SetContentHeight());
+            }
+        }
 
-    private void Update()
-    {
-#if UNITY_2017_2_OR_NEWER
-        if (!Application.isEditor && XRDevice.isPresent)
+        private IEnumerator SetContentHeight()
         {
-            floorHeightOffset = BoundaryManager.Instance.CurrentFloorHeightOffset;
+            if (frameWaitHack < 1)
+            {
+                // Not waiting a frame often caused the camera's position to be incorrect at this point. This seems like a Unity bug.
+                frameWaitHack++;
+                yield return null;
+            }
 
-            if (lastFloorHeight.y != floorHeightOffset)
+            if (XRDevice.GetTrackingSpaceType() == TrackingSpaceType.RoomScale)
             {
-                lastFloorHeight.y = floorHeightOffset;
-                transform.position = lastFloorHeight;
+                if (alignWithInitialHeadHeight)
+                {
+                    containerObject.position = new Vector3(containerObject.position.x, containerObject.position.y + CameraCache.Main.transform.position.y, containerObject.position.z);
```

Comment:
We could probably use a local Vector3 field and just reset it, then update it with the new position before assigning the container objects new position

Reply:
Would there be a benefit to that change?

Reply:
Removes the need for majick numbers and reduces the need to allocate a new Vector3

Reply:
Hmm true. As-is, this method is only called once, but I'll update this to future-proof it a bit.